### PR TITLE
Update feedback components for Chakra

### PR DESCRIPTION
**Pull request by wackerow**
## Description
- Update FeedbackWidget and FeedbackCard using Chakra-UI instead of emotion
- Patches FeedbackWidget button colors
- Patches FeedbackWidget to reinitialize component state when changing URL paths
- Patches FeedbackWidget to default keyboard focus to close button instead of "Yes" answer on widget open
- Update Nav icons to use IconButton

## Related Issue
#6374

### CODE REVIEW

File: src/components/FeedbackWidget.tsx [hunk 1]
```diff
@@ -17,157 +16,63 @@ import { translateMessageId } from "../utils/translations"
 import { useOnClickOutside } from "../hooks/useOnClickOutside"
 import { useKeyPress } from "../hooks/useKeyPress"
 import { useSurvey } from "../hooks/useSurvey"
+import { MdClose } from "react-icons/md"
```
Comment (pettinarip) [inline src/components/FeedbackWidget.tsx:19]: import order. Should be at the top with the external lib imports

File: src/components/NakedButton.tsx [hunk 2]
```diff
@@ -1,13 +1,24 @@
-import styled from "@emotion/styled"
+import React from "react"
+import { Button, ButtonProps } from "@chakra-ui/react"
+
+const NakedButton: React.FC<ButtonProps> = ({ children, ...props }) => (
```
Comment (pettinarip) [inline src/components/NakedButton.tsx:4]: Have you check the `ghost` variant? https://chakra-ui.com/docs/components/button/usage#button-variants

I think we should remove this component and start using `<Button variant="ghost">`. Any specific style we want to do on top of the default variant styles, we could add them on the Chakra theme definition as we did with the other variants.

Check the `variants` key in [here](https://github.com/ethereum/ethereum-org-website/blob/dev/src/%40chakra-ui/gatsby-plugin/components/Button.ts).

File: src/components/FeedbackWidget.tsx [hunk 3]
```diff
@@ -17,157 +16,63 @@ import { translateMessageId } from "../utils/translations"
...
-    transform: scale(1.1);
-    transition: transform 0.2s ease-in-out;
-  }
-  transition: transform 0.2s ease-in-out;
-`
-
-export interface IProps {
-  className?: string
+}
+const FixedDot: React.FC<FixedDotProps> = ({
```
Comment (pettinarip) [inline src/components/FeedbackWidget.tsx:24]: Just want to mention that there is a Chakra component called `IconButton` https://chakra-ui.com/docs/components/icon-button that perhaps could be a replacement for this. Not saying that this is in the scope of this PR but perhaps it is a better solution in the long term.

File: src/components/Layout.tsx [hunk 4]
```diff
@@ -213,7 +213,7 @@ const Layout: React.FC<IProps> = ({
               <VisuallyHidden isHidden={isZenMode}>
                 <Footer />
               </VisuallyHidden>
-              <FeedbackWidget />
+              <FeedbackWidget location={path} />
```
Comment (pettinarip) [inline src/components/Layout.tsx:216]: One common practice to deal with this is to do
```tsx
<FeedbackWidget key={path} />
```
that way, the component is going to remount everytime the path change and you will not have to do any manual reset in the inner component logic.

**Review by pettinarip — CHANGES_REQUESTED** [review]
Nicely done @wackerow left a few comments

File: src/components/FeedbackWidget.tsx (hunk 3 above)
Comment (wackerow) [reply src/components/FeedbackWidget.tsx:24]: Yes this component has more changes to come in a separate PR, thanks for noting!

File: src/components/FeedbackWidget.tsx [hunk 5]
```diff
@@ -220,82 +124,131 @@ const FeedbackWidget: React.FC<IProps> = ({ className }) => {
...
               fallbackFocus: `#dot`,
             }}
           >
-            <Container
-              bottomOffset={bottomOffset}
-              ref={containerRef}
-              className={className}
+            <Flex
               id="modal"
+              ref={containerRef}
```
Comment (pettinarip) [inline src/components/FeedbackWidget.tsx:148]: I think we should aim to use the Chakra Popover in this component. https://chakra-ui.com/docs/components/popover we will have better a11y and will abstract us for a bunch of things we are doing here (close icon, modal styles, focus trap, etc).

Not saying this should be in the scope of this PR but I'll create a new issue to tackle that after we merge this.

File: src/@chakra-ui/gatsby-plugin/components/Button.ts [hunk 6]
```diff
@@ -70,5 +70,19 @@ export const Button: ComponentStyleConfig = {
       color: "primary",
       borderColor: "primary",
     },
+    icon: {
+      appearance: "none",
+      display: "grid",
+      placeItems: "center",
```
Comment (pettinarip) [inline src/@chakra-ui/gatsby-plugin/components/Button.ts:76]: The button content is centered by default. If you inspect the [Button](https://chakra-ui.com/docs/components/button/usage) you will see that it is flex displayed and is centered.
```suggestion
```

File: src/components/Nav/index.tsx [hunk 7]
```diff
@@ -433,20 +415,28 @@ const Nav: React.FC<IProps> = ({ path }) => {
...
+                    as={isDarkTheme ? MdWbSunny : MdBrightness2}
+                    fontSize="2xl"
+                  />
+                }
+                variant="icon"
+                _hover={{ color: "primary" }}
+                onClick={toggleColorMode}
+              />
+              <ButtonLink to="/languages/" variant="icon">
+                <Flex flexWrap="nowrap" alignItems="center">
```
Comment (pettinarip) [inline src/components/Nav/index.tsx:433]: It is not semantically correct to place a div inside the button. I've removed it and is not wrapping.
```suggestion
```

File: src/components/FeedbackWidget.tsx [hunk 8]
```diff
@@ -220,82 +124,131 @@ const FeedbackWidget: React.FC<IProps> = ({ className }) => {
...
                   <Translation id="feedback-widget-thank-you-timing" />
-                </p>
+                </Text>
               )}
-              <ButtonContainer>
+              <Flex flexWrap="nowrap" gap={6} width="full">
                 {feedbackSubmitted ? (
                   <Button
-                    variant="outline-color"
+                    variant="solid"
```
Comment (pettinarip) [inline src/components/FeedbackWidget.tsx:218]: `solid` is the default variant, not need to define it.
```suggestion
```

**Review by pettinarip — CHANGES_REQUESTED** [review]
Nice! I think we can still remove some styles from our custom `icon` variant and fix the markdown semantics on the `Languages` button.

**Review by pettinarip — APPROVED** [review]
Awesome! 🚀 

I've just merged `dev` to fix merge conflicts 👍🏼

**Pull request merged**